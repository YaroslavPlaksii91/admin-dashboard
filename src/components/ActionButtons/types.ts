export type ActionButtonsProps = {
  onAddClick: () => void;
  onFilterClick: () => void;
  onSortClick: () => void;
  addButtonName: string;
};
