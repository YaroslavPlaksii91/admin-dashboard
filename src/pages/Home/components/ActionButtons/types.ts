export type ActionButtonsProps = {
  onAddClick: () => void;
  handleSort: (sortKey: string) => void;
  addButtonName: string;
  sortOptions: string[];
  filterValue: string;
  setFilterValue: (value: string) => void;
  filterTitle: string;
  filterOptions: string[];
};
