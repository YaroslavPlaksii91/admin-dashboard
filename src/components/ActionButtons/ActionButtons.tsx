import { FC } from 'react';
import { Box, Button, SvgIcon } from '@mui/material';

import { ActionButtonsProps } from './types';

export const ActionButtons: FC<ActionButtonsProps> = ({
  onSortClick,
  onFilterClick,
  onAddClick,
  addButtonName,
}) => {
  return (
    <Box sx={{ display: 'flex', mb: '47px', padding: '0 32px' }}>
      <Button
        variant="text"
        sx={{ color: 'grayDarkColor', mr: 8 }}
        onClick={onSortClick}
      >
        <SvgIcon titleAccess="Sort" fontSize="small" sx={{ mr: 2 }}>
          <use href="/src/assets/icons/sprite.svg#icon-sort"></use>
        </SvgIcon>
        Sort
      </Button>
      <Button
        variant="text"
        sx={{ color: 'grayDarkColor' }}
        onClick={onFilterClick}
      >
        <SvgIcon titleAccess="Filter" fontSize="small" sx={{ mr: 2 }}>
          <use href="/src/assets/icons/sprite.svg#icon-filter"></use>
        </SvgIcon>
        Filter
      </Button>
      <Button
        variant="text"
        sx={{ color: 'accentBlueColor', ml: 'auto' }}
        onClick={onAddClick}
      >
        <SvgIcon
          titleAccess="Add new ticket"
          sx={{ mr: 2, width: 8, height: 8 }}
        >
          <use href="/src/assets/icons/sprite.svg#icon-add"></use>
        </SvgIcon>
        {addButtonName}
      </Button>
    </Box>
  );
};
