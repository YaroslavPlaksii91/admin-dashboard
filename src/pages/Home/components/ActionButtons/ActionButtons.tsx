import { FC, useState, MouseEvent } from 'react';
import { Box, Button, SvgIcon } from '@mui/material';

import { ActionButtonsProps } from './types';
import {
  SORT_BTN_TEST_ID,
  FILTER_BTN_TEST_ID,
  ADD_BTN_TEST_ID,
} from './constants';
import { Popover } from '../Popover/Popover';
import { RadioButtons } from '../RadioButtons/RadioButtons';

export const ActionButtons: FC<ActionButtonsProps> = ({
  handleSort,
  onAddClick,
  addButtonName,
  sortOptions,
  setFilterValue,
  filterValue,
  filterTitle,
  filterOptions,
}) => {
  const [sortAnchorEl, setSortAnchorEl] = useState<HTMLButtonElement | null>(
    null,
  );
  const [filterAnchorEl, setFilterAnchorEl] =
    useState<HTMLButtonElement | null>(null);

  const handleSortClick = (event: MouseEvent<HTMLButtonElement>) => {
    setSortAnchorEl(event.currentTarget);
  };
  const handleFilterClick = (event: MouseEvent<HTMLButtonElement>) => {
    setFilterAnchorEl(event.currentTarget);
  };

  const handleSortClose = () => {
    setSortAnchorEl(null);
  };
  const handleFilterClose = () => {
    setFilterAnchorEl(null);
  };

  const sortOpen = Boolean(sortAnchorEl);
  const sortId = sortOpen ? 'sort-popover' : undefined;

  const filterOpen = Boolean(filterAnchorEl);
  const filterId = filterOpen ? 'filter-popover' : undefined;

  const handleFilterChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setFilterValue((event.target as HTMLInputElement).value);
  };

  return (
    <Box sx={{ display: 'flex', mb: '47px', padding: '0 32px' }}>
      <Button
        variant="text"
        sx={{ color: 'grayDarkColor', mr: 8 }}
        aria-describedby={sortId}
        onClick={handleSortClick}
        data-testid={SORT_BTN_TEST_ID}
      >
        <SvgIcon titleAccess="Sort" fontSize="small" sx={{ mr: 2 }}>
          <use href="/sprite.svg#icon-sort"></use>
        </SvgIcon>
        Sort
      </Button>

      <Popover
        id={sortId}
        open={sortOpen}
        anchorEl={sortAnchorEl}
        handleClose={handleSortClose}
      >
        {sortOptions.map(option => (
          <Button
            key={option}
            variant="text"
            sx={{ display: 'block', width: '100%', padding: '10px' }}
            onClick={() => handleSort(option)}
          >
            By {option}
          </Button>
        ))}
      </Popover>

      <Button
        variant="text"
        sx={{ color: 'grayDarkColor' }}
        aria-describedby={filterId}
        onClick={handleFilterClick}
        data-testid={FILTER_BTN_TEST_ID}
      >
        <SvgIcon titleAccess="Filter" fontSize="small" sx={{ mr: 2 }}>
          <use href="/sprite.svg#icon-filter"></use>
        </SvgIcon>
        Filter
      </Button>

      <Popover
        id={filterId}
        open={filterOpen}
        anchorEl={filterAnchorEl}
        handleClose={handleFilterClose}
      >
        <RadioButtons
          value={filterValue}
          handleChange={handleFilterChange}
          title={filterTitle}
          options={filterOptions}
        />
      </Popover>

      <Button
        variant="text"
        sx={{ color: 'accentBlueColor', ml: 'auto' }}
        onClick={onAddClick}
        data-testid={ADD_BTN_TEST_ID}
      >
        <SvgIcon titleAccess="Add new item" sx={{ mr: 2, width: 8, height: 8 }}>
          <use href="/sprite.svg#icon-add"></use>
        </SvgIcon>
        {addButtonName}
      </Button>
    </Box>
  );
};
