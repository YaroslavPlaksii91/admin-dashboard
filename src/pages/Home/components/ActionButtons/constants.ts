export const SORT_BTN_TEST_ID = 'sortButton';
export const FILTER_BTN_TEST_ID = 'filterButton';
export const ADD_BTN_TEST_ID = 'addButton';
