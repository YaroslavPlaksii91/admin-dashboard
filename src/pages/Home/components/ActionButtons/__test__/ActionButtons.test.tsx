import { cleanup, render, screen } from '@testing-library/react';

import { ActionButtons } from '../ActionButtons';

describe('ActionButtons component', () => {
  afterEach(cleanup);

  const mockProps = {
    handleSort: jest.fn(),
    onAddClick: jest.fn(),
    addButtonName: 'Add ticket',
    sortOptions: ['name', 'date', 'title'],
    setFilterValue: jest.fn(),
    filterValue: 'name',
    filterTitle: 'Name',
    filterOptions: ['high', 'low', 'normal'],
  };

  it('should be rendered', () => {
    const { asFragment } = render(<ActionButtons {...mockProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly', () => {
    render(<ActionButtons {...mockProps} />);

    const buttons = screen.getAllByRole('button');

    buttons.forEach(button => expect(button).toBeInTheDocument());
  });
});
