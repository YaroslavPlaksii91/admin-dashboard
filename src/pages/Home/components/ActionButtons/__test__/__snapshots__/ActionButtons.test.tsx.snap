// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButtons component should be rendered 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-la04gv"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-17u4eyv-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1v1n4yu-MuiSvgIcon-root"
        focusable="false"
        role="img"
        viewBox="0 0 24 24"
      >
        <use
          href="/src/assets/icons/sprite.svg#icon-sort"
        />
        <title>
          Sort
        </title>
      </svg>
      Sort
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-dpl43n-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1v1n4yu-MuiSvgIcon-root"
        focusable="false"
        role="img"
        viewBox="0 0 24 24"
      >
        <use
          href="/src/assets/icons/sprite.svg#icon-filter"
        />
        <title>
          Filter
        </title>
      </svg>
      Filter
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-e9wzt7-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1vh51qy-MuiSvgIcon-root"
        focusable="false"
        role="img"
        viewBox="0 0 24 24"
      >
        <use
          href="/src/assets/icons/sprite.svg#icon-add"
        />
        <title>
          Add new ticket
        </title>
      </svg>
      Add ticket
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
